### PR TITLE
[2109][7834101] VG No_PlainText_Secrets_In_Variables improvements

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.VariableGroup.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
             "Automated": "Yes",
             "MethodName": "CheckCredInVarGrp",
             "Rationale": "Keeping secrets such as connection strings, passwords, keys, etc. in plain text can expose the credentials to a wider audience and can lead to credential theft. Marking them as secret protects them from unitended disclosure and/or misuse.",
-            "Recommendation": "Refer: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=vsts&tabs=yaml%2Cbatch#secret-variables",
+            "Recommendation": "1. Navigate to the variable group --> 2. Goto variables --> 3. Lock the variables which contain secret using 'Change variable type to secret' option denoted by lock symbol against each variable.",
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.VariableGroup.json:49]: 2. Goto variables -> 2. Go to variables

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 2]
```diff
@@ -215,10 +218,13 @@ class VariableGroup: ADOSVTBase
                 {
                     $varList = @();
                     $noOfCredFound = 0;
-                    $patterns = $this.ControlSettings.Patterns | where {$_.RegexCode -eq "SecretsInBuild"} | Select-Object -Property RegexList;
+                    $patterns = $this.ControlSettings.Patterns | where-object {$_.RegexCode -eq "SecretsInVariables"} | Select-Object -Property RegexList;
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:221]: Make it array using @ and remove the  Measure-Object at 224

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 3]
```diff
@@ -215,10 +218,13 @@ class VariableGroup: ADOSVTBase
...
                     $exclusions = $this.ControlSettings.Build.ExcludeFromSecretsCheck;
+                    $exclusions += $this.ControlSettings.Release.ExcludeFromSecretsCheck; 
+                    $exclusions = @($exclusions | select-object -unique)
                     if(($patterns | Measure-Object).Count -gt 0)
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:225]: Remove measure-object as mentioned above

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 4]
```diff
@@ -252,10 +258,10 @@ class VariableGroup: ADOSVTBase
                         }
                         if($noOfCredFound -gt 0)
                         {
-                            $varList = $varList | select -Unique
-                            $controlResult.AddMessage([VerificationResult]::Failed, "Found secrets in variable group. Variables name: $varList" );
+                            $varList = @($varList | Select-Object -Unique)
+                            $controlResult.AddMessage([VerificationResult]::Failed, "Found secrets in variable group.`nList of variables: ", $varList );
                             $controlResult.SetStateData("List of variable name containing secret: ", $varList);
-                            $controlResult.AdditionalInfo += "Total number of variable(s) containing secret: " + ($varList | Measure-Object).Count;
+                            $controlResult.AdditionalInfo += "Count of variable(s) containing secret: " + ($varList | Measure-Object).Count;
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:264]: Measure-Object

Comment (SaikumarRagala) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:264]: At line 252, why are we adding double quotes like " $varName"?

**Review by SaikumarRagala — APPROVED** [review]

**Review by juhi037 — APPROVED** [review]

**Pull request merged**